### PR TITLE
Merge | Align Type aliases between netfx and netcore

### PR DESCRIPTION
**Pull request by MichelZ**
This PR solely aligns type aliases between netfx and netcore versions so they can be merged easier later.

Part of #2953

### CODE REVIEW

**Review by benrr101 — APPROVED** [review]
Looks good, but going forward, I think we can combine a couple of PRs like this together. I'm starting to get fatigue from reviewing these all independently 😅

**Comment by MichelZ** [comment]
> Looks good, but going forward, I think we can combine a couple of PRs like this together. I'm starting to get fatigue from reviewing these all independently 😅

Apologies, I really thought it makes it easier for review to have smaller, more targeted PR's... Guess I was wrong :)
Will combine them in the future

**Comment by benrr101** [comment]
There's definitely a balance :) I think we'll still take stuff when it's bite-sized like this, but it just might be slow with all the overhead. On the other hand, gigantic PRs are a lot more likely to just be outright rejected

**Review by mdaigle — APPROVED** [review]
Looks good after merge conflicts are fixed

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs [hunk 1]
```diff
@@ -10384,7 +10384,7 @@ internal Task TdsExecuteRPC(SqlCommand cmd, IList<_SqlRPC> rpcArray, int timeout
                                         else
                                         {
                                             Debug.Assert(value is SqlChars, "Unknown value for Ansi datatype");
-                                            s = new String(((SqlChars)value).Value);
+                                            s = new string(((SqlChars)value).Value)
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs:10387]:
```suggestion
                                            s = new string(((SqlChars)value).Value);
```

**Review by cheenamalhotra — CHANGES_REQUESTED** [review]

**Comment by MichelZ** [comment]
AZP needs a smack on the head

**Review by David-Engel — APPROVED** [review]

**Comment by MichelZ** [comment]
@cheenamalhotra Anything I can do to move this along? Thx

**Pull request merged**